src/instructlab/train/linux_train.py

```diff
@@ -96,6 +98,17 @@ def _gib(size: int) -> str:
             "stopping the server to free up about 5 GiB of GPU memory."
         )
 
+def determine_batch_size(device):
+    if device.type == "cpu":
+        # Gets the total system memory, excluding swap
+        mem_bytes = psutil.virtual_memory().total
+        mem_gb = mem_bytes / 1024 / 1024 / 1024
+        return 2 if mem_gb < 60 else 4
```

Comment:
What's the rationale behind a batch size of 2 or 4 for the CPU case?

Reply:
Larger batch sizes use more memory. It is attempting to fit as large of a size as possible.